spec/src/lib.rs

```diff
@@ -298,6 +316,23 @@ impl Genesis {
                 }
             }
         }
+
+        // Check system cells data hash
+        let check_data_hash = |(tx_index, output_index), hash: &H256| {
+            if data_hashes.get(&hash.pack()) != Some(&(tx_index, output_index)) {
+                return Err(format!(
+                    "Invalid output data for tx-index: {}, output-index: {}, expected data hash: {:x}",
+                    tx_index, output_index,
+                    hash,
+                ));
+            }
+            Ok(())
+        };
+        check_data_hash((0, 1), &CODE_HASH_SECP256K1_BLAKE160_SIGHASH_ALL)?;
```

Comment:
Why not define constants to represent the index of the certain system cells?

Reply:
Good catch